### ISSUE TITLE
Add model versioning with MLflow

### ISSUE THREAD

**Issue by MatthijsKeep** [MatthijsKeep/linear-issue-tracking#26]
Log architecture and params; experiment with variants (e.g., add batch norm) to learn hyperparameter sensitivity in isolation.
